src/Common/src/Interop/Unix/System.Net.Security.Native/Interop.GssApi.cs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Text;
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static class GssApi
+    {
+        internal static bool EstablishSecurityContext(
+            ref SafeGssContextHandle context,
+            SafeGssCredHandle credential,
+            bool isNtlm,
+            SafeGssNameHandle targetName,
+            Interop.NetSecurity.GssFlags inFlags,
+            byte[] buffer,
+            out byte[] outputBuffer,
+            out uint outFlags)
+        {
+            outputBuffer = null;
+            outFlags = 0;
+
+            if (context == null)
+            {
+                context = new SafeGssContextHandle();
+            }
+
+            SafeGssBufferHandle outputToken;
+            NetSecurity.Status status;
+            NetSecurity.Status minorStatus;
+            int outputLength;
+            status = NetSecurity.InitSecContext(out minorStatus,
+                                                credential,
+                                                ref context,
+                                                isNtlm,
+                                                targetName,
+                                                (uint)inFlags,
+                                                buffer,
+                                                (buffer == null) ? 0 : buffer.Length,
+                                                out outputToken,
+                                                out outputLength,
+                                                out outFlags);
+
+            using (outputToken)
+            {
+                if ((status != NetSecurity.Status.GSS_S_COMPLETE) && (status != NetSecurity.Status.GSS_S_CONTINUE_NEEDED))
+                {
+                    throw new NetSecurity.GssApiException(SR.net_context_establishment_failed, status, minorStatus);
+                }
+
+                outputBuffer = new byte[outputLength];
+                if (outputLength > 0)
+                {
+                    Interop.NetSecurity.CopyBuffer(outputToken, outputBuffer, 0);
```

Comment:
Change this to an instance method in SafeGssBufferHandle. Also take the length as a parameter and do the allocation inside the method. If we need to change later, it will be a single place. So the pseudocode I am suggesting is:
````
outputBuffer.= outputToken.CopyBuffer(outputLength);
````